### DISCUSSION TITLE
Suppressing trailing whitespace in multiline show-lambda rules for inline elements

### DISCUSSION THREAD

**Discussion by adiabatic** [typst/typst#3105] [Q&A] [answered]
Pardon the HTMLism that is "inline elements"; I'm not sure what the proper term is in Typst.

I'm trying to automatically highlight my raw text:

```typst
#show raw: (it) => [
  #highlight(it)
]

`this`: that
```

However, this introduces unwanted space after the highlighted raw text:
![Screenshot 2023-12-30 at 8 22 09 PM](https://github.com/typst/typst/assets/101246/e2581ecd-f037-4ae3-95ec-d23d8bf155ae)

If I remove the newline after the `highlight(it)` I don't get the space:

```typst
#show raw: (it) => [
  #highlight(it)]
```

…however, I'm not a fan of mixing C- and Lisp-style brace/bracket/paren closings in the same document.

Is there a way to close up the whitespace between the `#highlight()` call and the enclosing block while keeping a consistent source-code layout?

Thanks!

(This is a simplified example; my real show-lambda rule has other things in it, which is why I'm not doing the obvious workaround of putting it all on one line.)

**Comment by MDLC01** [comment]
Have you tried using braces instead of brackets? That way you have control over whitespace.

```typ
#show raw: (it) => {
  highlight(it)
}
```

You can explicitly insert whitespace when in code mode using `[ ]`.

**Reply by adiabatic** [reply]
Nope. Your suggestion does what I want…

```typst
#show raw: (it) => {
  set text(font: "Comic Sans MS")
  highlight(fill: rawBackgroundColor, it)
}
```

…but is there a way to have `it` be at the end of the function, all by itself, instead of being passed in to the last function? I tried stuff like

```typst
#show raw: (it) => {
  set text(font: "Comic Sans MS") // not the actual font used
  set highlight(fill: rawBackgroundColor)
  it
}
```

and the font changes properly, but the highlight doesn't get applied anymore.

**Reply by laurmaedje** [reply]
because `highlight` takes the to-be-highlighted content as an argument, such a set rule doesn't trigger it by itself. it only configures further uses of highlight. you can apply `highlight` to the full remaining block with an "everything show rule": `show: highlight`.

**Reply by MDLC01** [reply, marked as answer]
```typ
#show raw: it => {
  set text(font: "Comic Sans MS")
  show: highlight.with(fill: rawBackgroundColor)
  it
}
```
should be your final solution.

**Reply by adiabatic** [reply]
Thanks. I've been having trouble understanding how to choose what I want between set rules, show-set rules, show-lambda rules (a specific case of show-everything rules), and now using functions' `with` method with a (checks page) _show-everything_ rule. I have a `#show: columns.with(4)` line in my document, but I still haven't quite cracked the code, so to speak. All this brings me closer to actually understanding Typst — particularly @laurmaedje's comment. Thanks, everyone!
